script.js

```diff
@@ -53,81 +67,137 @@ my_widget_script =
           var fw = $('input[name^=fw]', tr);
-          if ($(this).val() && fw.val() && moles1.val())
-            equiv.val(($(this).val() / fw.val() / moles1.val()).toFixed(nFixed))
-        });
-        $('#the_form input[name^=fw]').on('keyup change', function() {
-          var tr = $(this).closest('tr');
           var amount = $('input[name^=amount]', tr);
-          var fw = $('input[name^=fw]', tr);
           var moles = $('input[name^=moles]', tr);
-          var equiv = $('input[name^=equivalents]', tr);
-          if (moles.val() && fw.val()) {
-            amount.val((moles.val() * fw.val()).toFixed(nFixed))
-          }
-        });
-        $('#the_form input[name^=fw]').on('change', function() {
-          var tr = $(this).closest('tr');
-          var amount = $('input[name^=amount]', tr);
-          var equiv = $('input[name^=equivalents]', tr);
-          if (!equiv.val()) {
-            amount.change();
-          }
-        });
-        $('#the_form input[name^=equivalents]').on('keyup change', function() {
-          var tr = $(this).closest('tr');
-          var amount = $('input[name^=amount]', tr);
-          var fw = $('input[name^=fw]', tr);
-          var moles = $('input[name^=moles]', tr);
-          var moles1 = $('#the_form input[name=moles1_formula]');
-          var equiv = $('input[name^=equivalents]', tr);
-          if (equiv.val() && fw.val() && moles1.val()) {
-            moles.val((moles1.val() * equiv.val()).toFixed(nFixed));
-            fw.change();
+
+          if (is_initial_row) {
+            if (fw.val() && !moles.val()) {
+              moles.val((amount.val() / fw.val()).toFixed(nFixed))
+              my_widget_script.enable_all_records()
+            } else if (fw.val() && amount.val() && moles.val()) {
+              moles.val((amount.val() / fw.val()).toFixed(nFixed))
+              my_widget_script.change_record_value()
+            }
+          } else {
+            var equiv = $('input[name^=equivalents]', tr);
+            var moles1 = $('#the_form input[name=moles1_number]');
+            if (fw.val()) {
+              moles.val((amount.val() / fw.val()).toFixed(nFixed));
+              equiv.val((amount.val() / fw.val() / moles1.val()).toFixed(nFixed))
+            }
           }
-        });
+        }),
+
+            $('#the_form input[name^=fw]').on('change', function() {
+              var tr = $(this).closest('tr');
+              console.log('In fw change handler '+ tr.attr('class'))
+              var is_initial_row = tr.attr('class') == 'initialRow';
+              var fw = $('input[name^=fw]', tr);
+              var amount = $('input[name^=amount]', tr);
+              var moles = $('input[name^=moles]', tr);
+              if (is_initial_row) {
+                if (amount.val() && !moles.val()) {
+                  moles.val((amount.val() / fw.val()).toFixed(nFixed))
+                } else if (moles.val() && !amount.val()) {
```

Comment:
What happens when initial fw is changed and both moles and amount are set? this case isn't handled. It should, IMO, result in amount being updated. So:
```suggestion
                } else if (moles.val()) {
```

Reply:
yes done now